### PR TITLE
Category

### PR DESCRIPTION
**Pull request by NikitaLafinskiy**

### CODE REVIEW

File: src/main/java/com/bookstore/entity/Book.java [hunk 1]
```diff
@@ -41,4 +47,10 @@ public class Book {
 
     @Column(columnDefinition = "TINYINT(1)")
     private boolean isDeleted = false;
+
+    @ManyToMany(fetch = FetchType.LAZY)
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/entity/Book.java]:
```suggestion
    @ManyToMany
```

File: src/main/java/com/bookstore/service/category/CategoryServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable).stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto findById(Long id) {
+        Category category = categoryRepository.findById(id)
+                .orElseThrow(EntityNotFoundException::new);
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/service/category/CategoryServiceImpl.java]: provide informative message with parameters

File: src/main/java/com/bookstore/service/category/CategoryServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryDto save(CreateCategoryRequestDto categoryDto) {
+        return categoryMapper.toDto(categoryRepository.save(categoryMapper.toEntity(categoryDto)));
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/service/category/CategoryServiceImpl.java]: let's improve readability
```suggestion
        Category category = categoryMapper.toEntity(categoryDto);
        return categoryMapper.toDto(categoryRepository.save(category));
```

File: src/main/java/com/bookstore/service/category/CategoryServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @Override
+    public CategoryDto update(Long id, CreateCategoryRequestDto categoryDto) {
+        Category category = categoryRepository.findById(id)
+                .orElseThrow(EntityNotFoundException::new);
+        categoryMapper.updateCategoryFromDto(categoryDto, category);
+        category = categoryRepository.save(category);
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/service/category/CategoryServiceImpl.java]:
```suggestion
        categoryRepository.save(category);
```

File: src/main/resources/db/changelog/changes/04-create-categories-table.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: name
+                  type: VARCHAR(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: description
+                  type: VARCHAR(9999)
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/04-create-categories-table.yaml]:
```suggestion
                  type: VARCHAR(255)
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by ahoienko — APPROVED** [review]
Good job!

**Pull request merged**